food.go

```diff
@@ -176,18 +175,15 @@ func (f *Food) Uninstall() error {
 }
 
 func unarchiveOrCopy(src, dest, urlPath string) error {
+	if err := archiver.Unarchive(src, dest); err == nil {
```

Comment:
how do we handle the case where the unarchiving process failed, like if the archive was corrupted, or the file extension did not match the archive format?

Reply:
Hm... We might be able to use the `ByExtension` function, like here: https://github.com/mholt/archiver/blob/2d14498067935bc817c9100490782583fb5ce44e/archiver.go#L198-L210

Reply:
Maybe something like this:

```go
if uaIface, err := archiver.ByExtension(source); err == nil {
	u, ok := uaIface.(Unarchiver)
	if !ok {
		return fmt.Errorf("format specified by source filename is not an archive format: %s (%T)", source, uaIface)
	}
	return u.Unarchive(source, destination)
}
```

or this:

```go
if uaIface, err := archiver.ByExtension(source); err == nil {
	if u, ok := uaIface.(Unarchiver); ok {
		return u.Unarchive(source, destination)
	}
}
```